server/src/test/java/org/candlepin/dto/api/v1/SubscriptionTranslatorTest.java

```diff
@@ -0,0 +1,169 @@
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import org.candlepin.dto.AbstractTranslatorTest;
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.shim.ProductDataTranslator;
+import org.candlepin.model.Cdn;
+import org.candlepin.model.Certificate;
+import org.candlepin.model.Owner;
+import org.candlepin.model.SubscriptionsCertificate;
+import org.candlepin.model.dto.ProductData;
+import org.candlepin.model.dto.Subscription;
+import org.candlepin.util.Util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+
+import static org.apache.commons.collections.CollectionUtils.isEmpty;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test suite for the ProductTranslator class
+ */
+public class SubscriptionTranslatorTest extends
+    AbstractTranslatorTest<Subscription, SubscriptionDTO, SubscriptionTranslator> {
+
+    private SubscriptionTranslator translator = new SubscriptionTranslator();
+
+    @Override
+    protected void initModelTranslator(ModelTranslator modelTranslator) {
+        modelTranslator.registerTranslator(new CdnTranslator(), Cdn.class, CdnDTO.class);
+        modelTranslator.registerTranslator(
+            new CertificateTranslator(), Certificate.class, CertificateDTO.class);
+        modelTranslator.registerTranslator(new NestedOwnerTranslator(), Owner.class, NestedOwnerDTO.class);
+        modelTranslator.registerTranslator(
+            new ProductDataTranslator(), ProductData.class, ProductDTO.class);
+        modelTranslator.registerTranslator(this.translator, Subscription.class, SubscriptionDTO.class);
+    }
+
+    @Override
+    protected SubscriptionTranslator initObjectTranslator() {
+        return this.translator;
+    }
+
+    @Override
+    protected Subscription initSourceObject() {
+        Subscription source = new Subscription();
+
+        source.setId("test_id");
+        source.setOwner(this.createOwner());
+        source.setProduct(this.createProduct());
+        source.setDerivedProduct(this.createProduct());
+        source.setProvidedProducts(this.createProducts());
+        source.setDerivedProvidedProducts(this.createProducts());
+        source.setQuantity(15L);
+        source.setStartDate(new Date());
+        source.setEndDate(new Date());
+        source.setContractNumber("test_contact");
+        source.setAccountNumber("test_acc_num");
+        source.setModified(new Date());
+        source.setOrderNumber("test_order_num");
+        source.setUpstreamPoolId("test_pool_id");
+        source.setUpstreamEntitlementId("test_ent_id");
+        source.setUpstreamConsumerId("test_cons_id");
+        source.setCertificate(this.createCert());
+        source.setCdn(this.createCdn());
```

Comment:
These 2 can be replaced by using `CertificateTranslatorTest.initSourceObject()` and `CdnTranslatorTest.initSourceObject()`

Reply:
Done.